apps/linear_algebra/src/blas_l1_generators.cpp

```diff
@@ -0,0 +1,218 @@
+#include <vector>
+#include "Halide.h"
+
+using namespace Halide;
+
+namespace {
+
+// Generator class for BLAS axpy operations.
+template<class T>
+class AXPYGenerator :
+      public Generator<AXPYGenerator<T> > {
+ public:
+  typedef Generator<AXPYGenerator<T> > Base;
+  using Base::target;
+  using Base::get_target;
+  using Base::natural_vector_size;
+
+  GeneratorParam<bool> assertions_enabled_ = {"assertions_enabled", false};
+  GeneratorParam<bool> use_fma_ = {"use_fma", false};
+  GeneratorParam<bool> vectorize_ = {"vectorize", true};
+  GeneratorParam<bool> parallel_ = {"parallel", true};
+  GeneratorParam<int>  block_size_ = {"block_size", 1024};
+  GeneratorParam<bool> scale_x_ = {"scale_x", true};
+  GeneratorParam<bool> add_to_y_ = {"add_to_y", true};
+
+  Param<T>   a_ = {"a", 1.0};
+  ImageParam x_ = {type_of<T>(), 1, "x"};
```

Comment:
The usual convention in Generator is to put ImageParams before Params. Is this different intentionally?